### DISCUSSION TITLE
Language support

### DISCUSSION THREAD

**Discussion by cyriltovena** [grafana/phlare#293] [Polls]
We're looking at adding more language support even though we may have support through some PPROF custom implementation we want to offer the best experience for all languages.

**Comment by daurnimator** [comment]
Would it be possible to gather traces from all native programs via reading dwarf info?
Same as you could attach to any running program with `perf`.

**Reply by cyriltovena** [reply]
Great idea something to investigate.

**Comment by JMLX42** [comment]
C++ ? :smiley:

**Reply by kawhicurry** [reply]
Wow, it must take a lot of time. I'm considering about compatibility. How about a library written with POSIX C? :)

**Reply by JMLX42** [reply]
> How about a library written with POSIX C?

@kawhicurry even better!

**Comment by mtrin** [comment]
Can it consume generic pprof coming from nodejs?
https://github.com/google/pprof-nodejs

**Reply by cyriltovena** [reply]
Yes it can already.

**Comment by sharozem** [comment]
Hi, is there any way to import profiling data from ruby on rails? Are there any gems currently available for this.

**Reply by cyriltovena** [reply]
Yes you can use the pyroscope SDK https://pyroscope.io/docs/ruby/

**Comment by hmehr** [comment]
No love for C++? 😄

**Reply by cyriltovena** [reply]
We're working on adding eBPF to Grafana Agent that might help.

hello from Grafana @hmehr hope you're doing well.

**Comment by multimac** [comment]
Was too excited to use Grafana Phlare for .NET and couldn't wait for upstream support, so I've made a small .NET program which collects CPU profiling via [dotnet-monitor](https://github.com/dotnet/dotnet-monitor) and converts it into the pprof format to be ingested via the Phlare agent.

https://github.com/multimac/dotnet-monitor-pprof-api

It's still a WIP, but I expect to start testing it in the development environments of my company in the coming weeks as I know there are a few others who are keen to get this kind of telemetry on our applications.

**Reply by cyriltovena** [reply]
Wow lovely @multimac You should try pyroscope .net SDK https://pyroscope.io/docs/dotnet/ you can ingest it now in Phlare and Pyroscope.
